packages/web/components/bridge/immersive/immersive-bridge.tsx

```diff
@@ -0,0 +1,257 @@
+import { Transition } from "@headlessui/react";
+import { BridgeTransactionDirection } from "@osmosis-labs/types";
+import { isNil } from "@osmosis-labs/utils";
+import { memo, PropsWithChildren, useState } from "react";
+import { useLockBodyScroll } from "react-use";
+
+import { Icon } from "~/components/assets";
+import { AmountScreen } from "~/components/bridge/immersive/amount-screen";
+import { AssetSelectScreen } from "~/components/bridge/immersive/asset-select-screen";
+import { Screen, ScreenManager } from "~/components/screen-manager";
+import { StepProgress } from "~/components/stepper/progress-bar";
+import { Button, IconButton } from "~/components/ui/button";
+import { EventName } from "~/config";
+import { BridgeFlowProvider } from "~/hooks/bridge";
+import { useAmplitudeAnalytics } from "~/hooks/use-amplitude-analytics";
+import { useDisclosure } from "~/hooks/use-disclosure";
+import { FiatRampKey } from "~/integrations";
+import { ModalCloseButton } from "~/modals";
+import { FiatOnrampSelectionModal } from "~/modals/fiat-on-ramp-selection";
+import { FiatRampsModal } from "~/modals/fiat-ramps";
+import { api } from "~/utils/trpc";
+
+enum ImmersiveBridgeScreens {
+  Asset = "0",
+  Amount = "1",
+  Review = "2",
+}
+
+const MemoizedChildren = memo(({ children }: PropsWithChildren) => {
+  return <>{children}</>;
+});
+
+export const ImmersiveBridgeFlow = ({
+  Provider,
+  children,
+}: PropsWithChildren<BridgeFlowProvider>) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [step, setStep] = useState<ImmersiveBridgeScreens>(
+    ImmersiveBridgeScreens.Asset
+  );
+  const [direction, setDirection] =
+    useState<BridgeTransactionDirection>("deposit");
```

Comment:
From other comment, IMO it's slightly more readable and less redirected if we just directly use the `"deposit" | "withdraw"` type literal